File: Form1.vb
Imports WindowsHookLib
Imports MouseSwap
Imports Microsoft.Win32

Public Class Form1
    Private MouseHooker As Inverter

    Private Sub Form1_Load(sender As System.Object, e As System.EventArgs) Handles MyBase.Load
        MouseHooker = New Inverter(False, False)
        AddHandler SystemEvents.DisplaySettingsChanged, AddressOf MsHook_Refresh
    End Sub


    Public Sub MsHook_Refresh()
        Dim mouseHooker As Inverter = Me.MouseHooker
        If Me.AutoFlip.Checked Then
            Dim screenOrientation As ScreenOrientation = SystemInformation.ScreenOrientation
            Select Case CInt(screenOrientation)
                Case 0
                    mouseHooker.Swap = False
                    mouseHooker.InvertX = False
                    mouseHooker.InvertY = False
                    GoTo Label_00FA

                Case 1
                    mouseHooker.Swap = True
                    mouseHooker.InvertX = True
                    mouseHooker.InvertY = False
                    GoTo Label_00FA

                Case 2
                    mouseHooker.Swap = False
                    mouseHooker.InvertX = True
                    mouseHooker.InvertY = True
                    GoTo Label_00FA

                Case 3
                    mouseHooker.Swap = True
                    mouseHooker.InvertX = False
                    mouseHooker.InvertY = True
                    GoTo Label_00FA
            End Select
            Return
        End If
        mouseHooker.Swap = Me.SwapXY.Checked
        mouseHooker.InvertX = Me.FlipX.Checked
        mouseHooker.InvertY = Me.FlipY.Checked
Label_00FA:
        If Me.Enabler.Checked Then
            If Not mouseHooker.Running Then
                mouseHooker.Start()
            End If
        ElseIf mouseHooker.Running Then
            mouseHooker.[Stop]()
        End If
        mouseHooker = Nothing
    End Sub

    Private Sub ParamUpdate(sender As System.Object, e As System.EventArgs) Handles SwapXY.CheckedChanged, FlipY.CheckedChanged, FlipX.CheckedChanged, Enabler.CheckedChanged, AutoFlip.CheckedChanged
        If Me.Visible Then MsHook_Refresh()
    End Sub
End Class

File: MouseSwapper/Form1.cs

using System;
using Microsoft.Win32;
using System.Windows.Forms;

public partial class Form1
{

	private Inverter MouseHooker;
	private void Form1_Load(object sender, System.EventArgs e)
	{
		MouseHooker = new Inverter();
		SystemEvents.DisplaySettingsChanged += DisplayUpdated;
	}

    void DisplayUpdated(object sender, EventArgs e)
    {
        ScrRot.SelectedIndex = (int)SystemInformation.ScreenOrientation;
        MsHook_Refresh(sender, e);
    }


    public void MsHook_Refresh(object sender, EventArgs e)
	{
        var hooker = this.MouseHooker;
        var screenOrientation = SystemInformation.ScreenOrientation;
        var targetOrient = (ScreenOrientation)ScrRot.SelectedIndex;
        if (screenOrientation != targetOrient)
        {
            if (ScrRot.SelectedIndex >= 0)
            {
                Display.Rotate(1, targetOrient);
            }
        }
        if (this.AutoFlip.Checked)
        {
            switch ((int)(screenOrientation))
            {
                case 0:
                    hooker.Set(false, false, false);
                    break;
                case 1:
                    hooker.Set(true, false, true);
                    break;
                case 2:
                    hooker.Set(true, true, false);
                    break;
                case 3:
                    hooker.Set(false, true, true);
                    break;
            }
        }
        else
        {
            hooker.Set(FlipX.Checked, FlipY.Checked, SwapXY.Checked);
        }
        if (Enabler.Checked)
        {
            if (!hooker._running)
                hooker.Start();
        }
        else if (hooker._running)
        {
            hooker.Stop();
        }
    }

    private void ParamUpdate(System.Object sender, System.EventArgs e)
	{
        if (Visible)    
    	    MsHook_Refresh(sender, e);
        FlipX.Enabled = FlipY.Enabled = SwapXY.Enabled = !AutoFlip.Checked;
    }

    public Form1()
	{
		Load += Form1_Load;
        InitializeComponent();
        ScrRot.SelectedIndex = (int)SystemInformation.ScreenOrientation;
	}
}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
